File: src/main/java/io/github/oleksivio/telegram/bot/api/annotations/filter/primitive/IntegerFilter.java
package io.github.oleksivio.telegram.bot.api.annotations.filter.primitive;

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface IntegerFilter {

    AnnotationState status() default AnnotationState.ON;

    int lessThan() default Integer.MAX_VALUE;

    int lessOrEqual() default Integer.MAX_VALUE;

    int moreThan() default Integer.MIN_VALUE;

    int moreOrEqual() default Integer.MIN_VALUE;

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/primitive/IntegerFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.primitive

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

/**
 * [IntegerFilter] annotation is used to filter incoming [Int] value
 *
 * @param lessThan - passes only less values
 *
 * @param lessOrEqual - passes  less and equal values
 *
 * @param greatThan - passes only great values
 *
 * @param greatOrEqual - passes great and equals values
 */
@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class IntegerFilter(
        val status: AnnotationState = AnnotationState.ON,
        val lessThan: Int = Integer.MAX_VALUE,
        val lessOrEqual: Int = Integer.MAX_VALUE,
        val greatThan: Int = Integer.MIN_VALUE,
        val greatOrEqual: Int = Integer.MIN_VALUE
)
